components/server/src/workspace/workspace-classes.ts

```diff
@@ -159,9 +159,13 @@ export namespace WorkspaceClasses {
      */
     export async function getConfiguredOrUpgradeFromLegacy(
         user: User,
+        project: Project | undefined,
         classes: WorkspaceClassesConfig,
         entitlementService: EntitlementService,
     ): Promise<string> {
+        if (project?.settings?.workspaceClasses?.regular) {
```

Comment:
Should the project preference hold over the `entitlement service` answer whether the user gets more resources?

Reply:
:dart: Yes. `entitlementService.userGetsMoreResources` is meant to guard whether we get anything besides "default".

Reply:
It is not used below for the user settings. so why would this be different for the project?

Reply:
Ah, misinterpreted the current code on multiple levels. You're right, Sven.

We do not check whether users are eligible on this level, also because we need to make up- and downwards migration through users choices and eligibility possible, including other constraints during elavation prebuilds -> regular workspaces.

This code is good as is, especially in the context of this PR. :+1: